lsif/src/server/middleware/errors.ts

```
@@ -9,11 +8,18 @@ import { pipeline as _pipeline } from 'stream'
  */
 export const errorHandler = (
     logger: Logger
-): ((error: any, req: express.Request, res: express.Response, next: express.NextFunction) => void) => (
-    error: any,
+): ((
+    error: Error & { status?: number },
```

In JS, you can throw non-Error values, so this should handle `unknown` (or `any`)

---

What's the idiomatic way to go from unknown to "something with a status property" or "something with a message" property?

---

I'm refactoring it to use a type guard (which I assume is correct here).

---

TS doesn't support narrowing with the `in` operator from the `unknown` type yet (https://github.com/microsoft/TypeScript/issues/21732), but a custom type guard `hasProperty()` would.

```ts
export const hasProperty<K extends string, T extends object>(key: K, value: T): value is T & Record<K, unknown> =>
  key in value
```